core/serix/map_decode.go

```diff
@@ -0,0 +1,372 @@
+package serix
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+	"unicode/utf8"
+
+	"github.com/pkg/errors"
+)
+
+func (api *API) mapDecode(ctx context.Context, mapVal any, value reflect.Value, ts TypeSettings, opts *options) error {
+	if err := api.mapDecodeBasedOnType(ctx, mapVal, value, value.Type(), ts, opts); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (api *API) mapDecodeBasedOnType(ctx context.Context, mapVal any, value reflect.Value,
+	valueType reflect.Type, ts TypeSettings, opts *options) error {
+	globalTS, _ := api.getTypeSettings(valueType)
+	ts = ts.merge(globalTS)
+	switch value.Kind() {
+	case reflect.Ptr:
+		if valueType == bigIntPtrType {
+			bigIntHexStr := mapVal.(string)
+			bigInt, err := DecodeUint256(bigIntHexStr)
+			if err != nil {
+				return errors.Wrap(err, "failed to read big.Int from map")
+			}
+			value.Addr().Elem().Set(reflect.ValueOf(bigInt))
+			return nil
+		}
+
+		elemType := valueType.Elem()
+		if elemType.Kind() == reflect.Struct {
+			if value.IsNil() {
+				value.Set(reflect.New(elemType))
+			}
+			elemValue := value.Elem()
+			return api.mapDecodeStruct(ctx, mapVal, elemValue, elemType, ts, opts)
+		}
+
+		if elemType.Kind() == reflect.Array {
+			if value.IsNil() {
+				value.Set(reflect.New(elemType))
+			}
+			sliceValue := sliceFromArray(value.Elem())
+			sliceValueType := sliceValue.Type()
+			if sliceValueType.AssignableTo(bytesType) {
+				innerTs, ok := api.getTypeSettings(valueType)
```

Comment:
Is the `valueType` correct here? Or does this have to be `sliceValueType`?
Just wondering, because its called `innerTs`

Reply:
`valueType` is the concrete type of the value where `sliceValueType` wouldn't map to get the actual type settings.